client/landing/gutenboarding/stores/onboard/resolvers.ts

```diff
@@ -21,7 +23,24 @@ import { receiveVerticals } from './actions';
  */
 export function* getVerticals() {
 	const url = 'https://public-api.wordpress.com/wpcom/v2/verticals';
-	const verticals = yield apiFetch( { url } );
+	let locale;
+	if ( ! locale && typeof navigator === 'object' && 'languages' in navigator ) {
+		for ( const langSlug of navigator.languages ) {
+			const language = getLanguage( langSlug.toLowerCase() );
+			if ( language ) {
+				locale = language.langSlug;
+				break;
+			}
+		}
+	}
```

Comment:
This is to check users browser language settings against [our supported languages](https://github.com/Automattic/wp-calypso/blob/16122154748a0656c36bd73cd2da9c1e8ff5c812/client/languages.js), and use that as the locale for the API request. Is that right?


@akirk is [`client/languages`](https://github.com/Automattic/wp-calypso/blob/16122154748a0656c36bd73cd2da9c1e8ff5c812/client/languages.js) something like the officially supported WordPress.com languages? If we make an API request with one of those languages as the locale, we might expect a localized response?

This loop allows us to find an appropriate language based on the user's browser with any number of fallbacks, as opposed to requesting in the user's preferred language and getting a response in the default (`en`) if it's not supported. Is that an accurate description?

We really don't need `getLanguage` for this, and we might consider isolating and publishing `client/languages` if it's useful for this type of thing.

x-ref relevant conversation: p1574068628441200-slack-luna

Reply:
Yes, these are the available languages. Responses are localized depending on the translation status. I wonder, though, if we need or should do the detection here. Traditionally the detection is done on the landing page https://wordpress.com/ that might redirect to localized landing pages which then link to the onboarding with the language as a parameter (e.g. https://wordpress.com/start/fr). We then use the `LocaleSuggestions` class to offer a switch into the right language.
Is there a particular reason from deviating from this previous implementation?